test/profiler/test_profiler.py

```diff
@@ -127,6 +127,16 @@ def custom_layer(input_ten):
             q = s.sum()
             q.backward()
 
+        # Only testing that emit_itt runs when
+        # record_shapes option is enabled.
+        with torch.autograd.profiler.emit_itt(record_shapes=True) as prof:
```

Comment:
@malfet This should be run as a separate test with a proper tag for ITT. Could you revert this PR? My PR seems to be blocked https://github.com/pytorch/pytorch/pull/82154

Reply:
Hi, may I have more explanations? Why this blocks your test?
This follows the [nvtx test](https://github.com/jingxu10/pytorch/blob/c6e22c244964e21648e5153c9119c0511946fbbf/test/profiler/test_profiler.py#L122-L128) above. Does the nvtx one block your test as well?

Reply:
You should have set something like below. I do not think default flow support ITT, but if I rebased onto master, your code are running and generated an error `RuntimeError: Can't use ITT profiler - PyTorch was compiled without ITT`. Also, I believe you should have written a separate unit test to set something like below. CUDA available is not meaning ITT available.
```
@unittest.skipIf(not HAS_PSUTIL, "Requires psutil to run")
@unittest.skipIf(TEST_WITH_ASAN, "Cannot test with ASAN")
@unittest.skipIf(IS_WINDOWS, "Test is flaky on Windows")
@unittest.skipIf(not torch.cuda.is_available(), "CUDA is required")
```

Reply:
Oh, I see. Should I file a separate PR today to get this fixed?

Reply:
yes, that would be another solution. Let us fix this ASAP!

Reply:
@slgong-fb @malfet PR filed at https://github.com/pytorch/pytorch/pull/86174

Reply:
I agree it would be nice for this PR to be a separate test case and have `@unittest.skipIf(not HAS_PSUTIL, "Requires psutil to run")`, but not sure about other decorators, as test does not seem to fail neither on Windows nor on Mac right now, does it?

Reply:
@slgong-fb @malfet Add itt availability check when performing itt ut at https://github.com/pytorch/pytorch/pull/86199. ITT tests will be skiped if itt is not available.